### ISSUE TITLE
prefer-spread on strings not supported by all compile targets

### ISSUE THREAD

**Issue by TSMMark** [sindresorhus/eslint-plugin-unicorn#2028]
prefer-spread on strings not supported by all compile targets.

`prefer-spread` rule maybe should have an option to disable for string split

```
Prefer the spread operator over `String#split('')`
```

In TS you get an error. Forcing `downlevelIteration` is not okay because it increases bundle size.

```
Type 'string' can only be iterated through when using the '--downlevelIteration' flag or with a '--target' of 'es2015' or higher.ts(2802)
```

However, even `"target": "es2015"` and `"target": "es6"` do not work because string spread was not added til `es2018` I think

```js
[...'abcdefg']
```

**Comment by sindresorhus** [comment]
We are not going to add any option for this. Our rules optimize for modern targets. You can disable the rule if it doesn't work for you.

**Comment by fregante** [comment]
I see several issues here:

> because it increases bundle size

It literally adds 500 bytes per bundle (minified)

> string spread was not added til `es2018` I think

That's not right and that's not what the compiler suggests. See https://node.green/#ES2015-syntax-spread-syntax-for-iterable-objects-with-strings--in-function-calls

I think you're confusing it with object spread `{...obj}`

---

If you're seeing that error, you're likely targeting IE11; a 500-byte addition is the least of your worries here.

If you're not, then you should ensure you use at least es2015 as the target

**Comment by TSMMark** [comment]
> That's not right and that's not what the compiler suggests

I know the error says es2015 should allow it — but it didn't make a difference when I tried it. That's why I mentioned it.
